### PR TITLE
Zh 3272

### PR DESCRIPTION
**Pull request by weakish**

### CODE REVIEW

**Comment by leeyeh** [comment]
Deploy preview for *lc-docs-en* ready!

Built with commit a5f1b7fc908c8745a2c1512db3f558b79867b8fb

https://deploy-preview-92--lc-docs-en.netlify.com

File: views/realtime-guide-intermediate.md [hunk 1]
```diff
@@ -285,36 +287,48 @@ do {
...
 });
 ```
 ```cs
-await conversation.RecallAsync(message);
+var newMessage = new AVIMTextMessage("The new message.");
+await conversation.UpdateAsync(oldMessage, newMessage);
 ```
 
-After Tom calls `recallMessage`, other members in the conversation will receive the `MESSAGE_RECALL` event:
+If the modification succeed, other members in the conversation will receive a `MESSAGE_UPDATE` event:
```
Comment (ghost) [inline views/realtime-guide-intermediate.md]: succeed -> succeeded

File: views/realtime-guide-intermediate.md [hunk 2]
```diff
@@ -350,37 +362,44 @@ func client(_ client: IMClient, conversation: IMConversation, event: IMConversat
...
+For Android and iOS SDKs, if caching is enabled (enabled by default), the SDKs will first update the modified message in the cache and then trigger an event to the app. When you receive such event, simply refresh the chatting page to reflect the latest collection of messages.
+
+If a message is modified by the system (for example, due to keyword filtering or by a hook on LeanEngine), the sender will receive a `MESSAGE_UPDATE` event, and other members in the conversation will receive the modified message.
+
+### Recall a Message
 
-Beside recalling the message, Tom can also **edit the message directly**. When this happens, what you would do is not to update the original message instance, but to create a new one and call `Conversation#updateMessage(oldMessage, newMessage)` to submit the request to the cloud. Here is a code example:
+Besides modifying an sent message, a user can also recall a message they sent.
```
Comment (ghost) [inline views/realtime-guide-intermediate.md]: an sent message -> a sent message

File: views/realtime-guide-intermediate.md [hunk 3]
```diff
@@ -390,48 +409,36 @@ do {
...
 });
 ```
 ```cs
-var newMessage = new AVIMTextMessage("The new message.");
-await conversation.UpdateAsync(oldMessage, newMessage);
+await conversation.RecallAsync(message);
 ```
 
-After Tom updates the message, other members in the conversation will receive the `MESSAGE_UPDATE` event:
+Once recalling a message succeed, other members in the conversation will receive the `MESSAGE_RECALL` event:
```
Comment (ghost) [inline views/realtime-guide-intermediate.md]: succeed > succeeded

**Review by ghost — CHANGES_REQUESTED** [review]

**Review by ghost — APPROVED** [review]

**Pull request merged**